agreementbot/agreementbot.go

```diff
@@ -797,14 +797,22 @@ func DeleteMessage(msgId int, agbotId, agbotToken, exchangeURL string, httpClien
 // microservices. If the agbot is working with a policy file that was generated from a pattern, then it will do searches
 // by pattern. If the agbot is working with a manually created policy file, then it will do searches by list of
 // microservices.
-func (w *AgreementBotWorker) searchExchange(pol *policy.Policy, searchOrg string) (*[]exchange.SearchResultDevice, error) {
+func (w *AgreementBotWorker) searchExchange(pol *policy.Policy, polOrg string) (*[]exchange.SearchResultDevice, error) {
 
 	// If it is a pattern based policy, search by worload URL and pattern.
 	if pol.PatternId != "" {
+		// get a list of node orgs that agbot is serving for this pattern
+		nodeOrgs := w.PatternManager.GetServedNodeOrgs(polOrg, exchange.GetId(pol.PatternId))
+		if len(nodeOrgs) == 0 {
```

Comment:
If the agbot is not configured to serve any specific node orgs, then isnt it always serving the same org that the pattern is in, by default?

Reply:
I'm unclear about how we get to this situation. I did tell ling that when running the `/orgs/{orgid}/agbots/{id}/patterns` exchange api that if they omit `nodeOrgids` in the body, the exchange will automatically set it to the pattern org. But if they explicitly set `"nodeOrgids": [ ]` i currently don't change it, i just save that. Is this the case we have here?

Reply:
This is not the case where nodeOrgid is omitted or empty in the `POST /orgs/{orgid}/agbots/{id}/patterns` call. It is the case when the user has a pattern but forgot to call `POST/orgs/{orgid}/agbots/{id}/patterns` for this pattern.

Reply:
I'm confused, if the user didn't call `POST/orgs/{orgid}/agbots/{id}/patterns`, how does the agbot even know about this pattern and have a policy file for it?

Reply:
It could be a left over from last time when deletion failed. It could be user copied it manually. Our code has to guard for this cases.

Reply:
In an offline discussion ling clarified to me that the `if len(nodeOrgs) == 0 {` should never happen, because `PatternManager.GetServedNodeOrgs()` will default `nodeOrgs` to the pattern org. But `if len(nodeOrgs) == 0 {` is here as a safeguard anyway so that this code doesn't have to know about that subtle behavior inside `PatternManager.GetServedNodeOrgs()`. That's a good practice, so i'm resolving this conversation.